### ISSUE TITLE
Rest-server  and webportal login not working on Azure VMs

### ISSUE THREAD

**Issue by iamzoumao** [Microsoft/pai#1309]
### Problem Statement
Hi guys, I have tried to deploy Open-PAI on Azure VM instances (tried both single-box deployment and deployment on a cluster of Azure GPU VMs). Unfortunately in both cases I encounter the same problem:
* I cannot login in to the webportal, after I input the default username (admin) and password (admin-password), nothing happens
* The rest-sever container reports "NotFoundError: API / is not found."
<img width="934" alt="1" src="https://user-images.githubusercontent.com/5267490/45264585-832f0100-b471-11e8-9581-71f19a1f6869.PNG">
<img width="588" alt="2" src="https://user-images.githubusercontent.com/5267490/45264631-1d8f4480-b472-11e8-9780-bcc2e8a0ecee.PNG">

### Open-PAI post-installation status
I use the single-box deployment as an example. All the descriptions apply for deployment on a cluster of Azure VMs.
* Kubernetes and all the service containers work well:
<img width="740" alt="3" src="https://user-images.githubusercontent.com/5267490/45264685-2e8c8580-b473-11e8-8025-256adedfddbc.PNG">

* HDFS work well.
* drivers-one-shot container work well: can run nvidia-smi correctly
* the end-to-end-test-deployment passed (although I find the tests dubious after checking the scripts)
![image](https://user-images.githubusercontent.com/5267490/45264741-c8543280-b473-11e8-9176-2cd3b7b1c51b.png)

### Pre-installation configuration
Basically I just follow the guides on github.
* docker pull the dev-box image, git update to the latest openpai release version;
* all the node(s) are standard Azure NC6 instances with 1 Tesla K80, no docker nor nvidia drivers pre-installed;
* modify the ~/pai-config/cluster-configuration.yaml slightly to match my environment
![image](https://user-images.githubusercontent.com/5267490/45264798-de162780-b474-11e8-9c7b-ad258eb01d41.png)

* paictl.py report to have started up all ther services successfully.

**Comment by Gerhut** [comment]
Hi @iamzoumao 

It looks like an error message returned by proxy. Are you requesting the web portal behind a proxy or in a VPN?

**Comment by iamzoumao** [comment]
Hi @Gerhut ,
Since the master node is a remote Azure VM instance, I am accessing the web portal on my local computer by ssh port forwarding:
* ssh -N -L localhost:9286:localhost:9286 username@master-node-dns-address
* visit localhost:9286 in a local chrome browser

May this cause the problem? Thank you!

**Comment by Gerhut** [comment]
You may try forward port 80 instead of 9286, hope it helps

Regards,
Qixiang

________________________________
收件人: Zou Mao <notifications@github.com>
发送时间: 星期一, 九月 10, 2018 12:49 下午
收件人: Microsoft/pai
抄送: Qixiang Cheng; Mention
主题: Re: [Microsoft/pai] Webportal login not working on Azure VMs (#1309)


Hi @Gerhut<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGerhut&data=02%7C01%7Cqixiang.cheng%40microsoft.com%7C9d67abe96a76476949c108d616d8c522%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636721517606941957&sdata=s5JMcpawtOXWhIIcyUzbuNWOvWd7scshGNUtdhVu4L4%3D&reserved=0> ,
Since the master node is a remote Azure VM instance, I am accessing the web portal on my local computer by ssh port forwarding:

  *   ssh -N -L localhost:9286:localhost:9286 username@master-node-dns-address
  *   visit localhost:9286 in a local chrome browser

May this cause the problem? Thank you!

―

**Comment by iamzoumao** [comment]
@Gerhut Unfortunately forwarding port 80 didn't help. 
I tried another approach: instead of ssh port forwarding, I modified the VM's inbound security rule to explictly allow TCP connections to all ports. However, the webportal still does not function correctly.

The dashboard:
![image](https://user-images.githubusercontent.com/5267490/45297884-547c5d80-b539-11e8-8b27-ec09e086d7af.png)
It says the connection to 10.0.13.5 (the private intranet IP of the VM) timeout

And the "jobs" page hangs here:
![image](https://user-images.githubusercontent.com/5267490/45297863-47f80500-b539-11e8-90e2-a16a3981849e.png)

Perhaps this is due to the webportal keeping connecting to the private intranet IP of the node?

**Comment by iamzoumao** [comment]
After some searches, I guess #1136 might be relevant, in which case the nginx configuration needs to be modified, and #1119 in v0.7.1 release should be included.

However, in my case, I am indeed using the latest v0.7.1 release, and Azure handles the port mapping for me, thus I'm not sure what I should do to fix the problem.

**Comment by Gerhut** [comment]
May I confirm that you are forwarding port 80 of the remote machine outside?

#1136 is that case that proxy the inner port outside with modifying the HTTP `host` header. Port forwarding is not related to that.

And may I confirm that the `http://localhost:80/` page is requesting data from `http://localhost:80/grafana` but not `http://inner-host/grafana` or `http://inner-host:3000/` if you forward ports by `ssh -N -L localhost:80:localhost:80 username@master-node-dns-address`?

**Comment by NothingLZ** [comment]
Hello, I have encountered some problems in the Azure VM deployment, can you add QQ（354020912（answer：罗增）） ?

**Comment by fanyangCS** [comment]
Hi @iamzoumao and @NothingLZ , please provide more information?
